nilearn/datasets/struct.py

```diff
@@ -106,38 +114,158 @@ def fetch_icbm152_2009(data_dir=None, url=None, resume=True, verbose=1):
     return Bunch(**params)
 
 
-def load_mni152_template():
-    """Load skullstripped 2mm version of the MNI152 originally distributed
-    with FSL.
+def load_mni152_template(resolution=2):
+    """This function takes the skullstripped 1mm-resolution version of the
+    MNI152 T1 template, originally distributed with FSL, and re-samples it
+    using a different resolution, if specified.
 
     For more information, see :footcite:`FONOV2011313`,
     and :footcite:`Fonov2009`.
 
+    Parameters
+    ----------
+    resolution: int, optional, Default = 2
+        If resolution is different from 1, the template is re-sampled with the
+        specified resolution.
+
+        .. versionadded:: 0.8.1
+
     Returns
     -------
-    mni152_template : nibabel object corresponding to the template
+    mni152_template : Nifti1Image, image representing the re-sampled
+        whole-brain template
+
+    See Also
+    --------
+    nilearn.datasets.load_mni152_gm_template : for details about version of the
+        MNI152 grey-matter template.
+
+    nilearn.datasets.load_mni152_wm_template : for details about version of the
+        MNI152 white-matter template.
 
     References
     ----------
     .. footbibliography::
 
     """
-    return check_niimg(MNI152_FILE_PATH)
 
+    brain_template = check_niimg(MNI152_FILE_PATH)
+    new_brain_template = new_img_like(
+        brain_template, get_data(brain_template).astype('uint16'))
```

Comment:
I had to do the conversion because of an error I previously reported [here](https://github.com/nilearn/nilearn/pull/2738#issuecomment-861892264) and [here](https://github.com/nilearn/nilearn/pull/2738#issuecomment-862430750). Could you advise how I shall solve it in a different way?

Reply:
I see, in that case maybe add a comment explaining that? But I think we may want to convert to float and rescale to [0, 1] anyway, for easier selection of thresholds, so this issue might disappear

Reply:
It solved, indeed.